### DISCUSSION TITLE
GHAS Certification Exam Prep: Part Four - Managing Your GitHub Securely

### DISCUSSION THREAD

**Discussion by queenofcorgis** [community/community#139526] [Enterprise]
_This discussion and  some of its comments have been edited and modified from its original format to enable evergreen learning_

This is our last part of [GHAS certification exam prep materials](https://github.com/orgs/community/discussions/137197), if you missed [Part One’s Discussion](https://github.com/orgs/community/discussions/137620), [Part Two’s Discussion](https://github.com/orgs/community/discussions/138220), and/or, [Part Three’s Discussion](https://github.com/orgs/community/discussions/138831) don’t worry feel free to get started and join us when you are caught up. The final installment covers Admin Essentials. 

Plus, prep materials and test questions to continue on studying.  

## Step One: Prep 📚
Use these materials to study before answering this week’s prep questions. 
* Microsoft Learn Modules
   * [GitHub administration for GitHub Advanced Security](https://learn.microsoft.com/en-us/training/modules/github-administration-github-advanced-security/)
   * [Manage sensitive data and security policies within GitHub](https://learn.microsoft.com/en-us/training/modules/manage-sensitive-data-security-policies/)
* Study guides from @ladykerr 
   * [Describe GitHub Advanced Security best practices](https://github.com/LadyKerr/github-certification-guide/blob/main/study-guides/gh-advanced-security.md#domain-6-describe-github-advanced-security-best-practices-results-and-how-to-take-corrective-measures)
* Videos 
    * GitHub Advanced Security - Pass the Exam from freeCodeCamp
       * [GHAS Best Practices](https://www.youtube.com/watch?v=i740xlsqxEM&t=8260s)
       * [GHAS for Enterprise](https://www.youtube.com/watch?v=i740xlsqxEM&t=8974s)
* Community member materials
  *  @Salgado2004's breakdown in [their comment](https://github.com/orgs/community/discussions/139526#discussioncomment-10778961)

### Question One:  How can you manage GitHub Advanced Security alerts?

A) By using the GitHub Advanced Security API endpoints
B) By manually checking each repository
C) By using a third-party tool
D) By disabling security features

### Question Two: What can you do to ensure that everyone in your organization is using GitHub Advanced Security?
A) Give access to security alerts to users and teams with write permission on their repositories.
B) Add a SECURITY.md file to the root, docs, or .github folder of your organization's repositories.
C) Set a security policy at the organization level.

### Question Three: Which GitHub Advanced Security feature isn't available on public repositories?
A) Secret scanning.
B) Security Overview.
C) Code scanning.

### Question Four: Which tool helps identify and block secrets from being pushed into repositories? 
A) Code scanning
B) Secret scanning
C) Dependency management
D) Issue tracking

### Question Five: How can security policies be managed within GitHub? 
A) By creating documentation that details security guidelines
B) By setting permissions and rules
C) By automating processes to prevent security breaches
D) All of the above

### Question Six: Which of the following is NOT a step in setting security policies within GitHub? 
A) Creating documentation that details security guidelines
B) Setting permissions and rules
C) Automating processes to prevent security breaches
D) Allowing unrestricted access to all users

### Question Seven: How can you ensure compliance with industry standards when managing sensitive data in GitHub? 
A) By ignoring security alerts
B) By using GitHub's basic security tools like secret scanning and code scanning
C) By disabling security features
D) By allowing bulk export of sensitive data

_[View the answers in this comment](https://github.com/orgs/community/discussions/139526#discussioncomment-10778893)_ 🧠

**_Use the discussion below to share additional study resources and respond to our prep questions_**

**Comment by jccampanero** [comment]
I believe these are the answers for this week:

**Question One: How can you manage GitHub Advanced Security alerts?**
**`A) By using the GitHub Advanced Security API endpoints`**
B) By manually checking each repository
C) By using a third-party tool
D) By disabling security features

**Question Two: What can you do to ensure that everyone in your organization is using GitHub Advanced Security?**
A) Give access to security alerts to users and teams with write permission on their repositories.
B) Add a SECURITY.md file to the root, docs, or .github folder of your organization's repositories.
**`C) Set a security policy at the organization level.`**

**Question Three: Which GitHub Advanced Security feature isn't available on public repositories?**
A) Secret scanning.
**`B) Security Overview.`**
C) Code scanning.

**Question Four: Which tool helps identify and block secrets from being pushed into repositories?**
A) Code scanning
**`B) Secret scanning`**
C) Dependency management
D) Issue tracking

**Question Five: How can security policies be managed within GitHub?**
A) By creating documentation that details security guidelines
B) By setting permissions and rules
C) By automating processes to prevent security breaches
**`D) All of the above`**

**Question Six: Which of the following is NOT a step in setting security policies within GitHub?**
A) Creating documentation that details security guidelines
B) Setting permissions and rules
C) Automating processes to prevent security breaches
**`D) Allowing unrestricted access to all users`**

**Question Seven: How can you ensure compliance with industry standards when managing sensitive data in GitHub?**
A) By ignoring security alerts
**`B) By using GitHub's basic security tools like secret scanning and code scanning`**
C) By disabling security features
D) By allowing bulk export of sensitive data

Thank you very much for the new course content this week.

Overall, thank you very much for this course. It has been a fantastic learning resource.

And thank you very much for your comments and your kindness, @queenofcorgis, @gitstua.

**Reply by Pagalkhan68** [reply]
PRAY AND HOPE THAT IF GOD HAS BOUGHT ME TO IT THAN IN JESUS NAME Y REDEEMER SHALL PULL ME THROUGH IT MY PERSONAL IDENTIFIABLE INFORMATION US CITIZEN SSN HOLDERS ACTS PRIVATE SECTORS DOC CDC PIA PG&E BETWEEN THE LINES ACTS AS WRITTEN VICTORY ALPHA HOLY BIBLE AMEN.

**Reply by queenofcorgis** [reply]
@jccampanero thank you so much for participating - I'm glad you enjoyed the course and asked questions ❤️

**Reply by jccampanero** [reply]
On the contrary, thank you @queenofcorgis Lili for your excellent work and your kindness.

**Comment by ghost** [comment]
Thanks again for the awesome list of resources, especially the Microsoft Learn paths! As for the answers, 

Q1. A
Q2. C
Q3. B
Q4. B
Q5. D
Q6. D
Q7. B

**Comment by Yordan-Ramchev** [comment]
1. A) By using the GitHub Advanced Security API endpoints
2. C) Set a security policy at the organization level
3. B) Security Overview
4. B) Secret scanning
5. D) All of the above
6. D) Allowing unrestricted access to all users
7. B) By using GitHub's basic security tools like secret scanning and code scanning

**Reply by Joooojj** [reply]
7

**Comment by geopd** [comment]
1. A
2. C
3. B
4. B
5. D
6. D
7. B

**Comment by profile-palash** [comment]
Question 1: How can you manage GitHub Advanced Security alerts? 

Answer: A. By using the GitHub Advanced Security API endpoints. <details> <summary>Explanation</summary> The GitHub Advanced Security API endpoints allow you to programmatically manage security alerts across your repositories. This includes retrieving, dismissing, and resolving alerts, which can help automate and streamline your security management processes. </details>

Question 2: What can you do to ensure that everyone in your organization is using GitHub Advanced Security? 

Answer: C. Set a security policy at the organization level. <details> <summary>Explanation</summary> Setting a security policy at the organization level ensures that all repositories within the organization adhere to the same security standards. This helps in maintaining consistency and compliance across the board. </details>

Question 3: Which GitHub Advanced Security feature isn’t available on public repositories? 

Answer: B. Security Overview. <details> <summary>Explanation</summary> The Security Overview feature, which provides a high-level view of security alerts and vulnerabilities across multiple repositories, is not available for public repositories. This feature is designed for private repositories to help organizations manage their security posture. </details>

Question 4: Which tool helps identify and block secrets from being pushed into repositories? 

Answer: B. Secret scanning. <details> <summary>Explanation</summary> Secret scanning is a feature that scans your repositories for sensitive data such as API keys, tokens, and passwords. It helps prevent these secrets from being accidentally committed to your codebase, thereby reducing the risk of security breaches. </details>

Question 5: How can security policies be managed within GitHub? 

Answer: D. All of the above. <details> <summary>Explanation</summary> Managing security policies within GitHub involves creating documentation that details security guidelines, setting permissions and rules to control access, and automating processes to prevent security breaches. These combined efforts help ensure a robust security posture. </details>

Question 6: Which of the following is NOT a step in setting security policies within GitHub? 

Answer: D. Allowing unrestricted access to all users. <details> <summary>Explanation</summary> Allowing unrestricted access to all users is not a step in setting security policies. Instead, security policies should focus on creating documentation, setting permissions and rules, and automating processes to enhance security. </details>

Question 7: How can you ensure compliance with industry standards when managing sensitive data in GitHub? 

Answer: B. By using GitHub’s basic security tools like secret scanning and code scanning. <details> <summary>Explanation</summary> Using GitHub’s basic security tools such as secret scanning and code scanning helps ensure compliance with industry standards by identifying and mitigating potential security risks in your codebase. </details>

**Comment by davevad93** [comment]
<details>
  <summary><h3>Question 1: How can you manage GitHub Advanced Security alerts?</h3></summary>
  <p><b>A) By using the GitHub Advanced Security API endpoints<b></p>
</details>

<details>
  <summary><h3>Question 2: What can you do to ensure that everyone in your organization is using GitHub Advanced Security?</h3></summary>
  <p><b>C) Set a security policy at the organization level<b></p>
</details>

<details>
  <summary><h3>Question 3: Which GitHub Advanced Security feature isn’t available on public repositories?</h3></summary>
  <p><b>B) Security Overview.<b></p>
</details>

<details>
  <summary><h3>Question 4: Which tool helps identify and block secrets from being pushed into repositories?</h3></summary>
  <p><b>B) Secret scanning<b></p>
</details>

<details>
  <summary><h3>Question 5: How can security policies be managed within GitHub?</h3></summary>
  <p><b>D) All of the above<b></p>
</details>

<details>
  <summary><h3>Question 6: Which of the following is NOT a step in setting security policies within GitHub?</h3></summary>
  <p><b>D) Allowing unrestricted access to all users<b></p>
</details>

<details>
  <summary><h3>Question 7: How can you ensure compliance with industry standards when managing sensitive data in GitHub?</h3></summary>
  <p><b>B) By using GitHub's basic security tools like secret scanning and code scanning<b></p>
</details>

**Reply by davevad93** [reply]
I've been inspired by @profile-palash format style!

**Reply by profile-palash** [reply]
@davevad93 Wow! You've enhanced readability by abstracting the answers 😍

**Comment by ghost** [comment]
Mb

**Comment by ghost** [comment]
> Mb

**Comment by juanlujand** [comment]
Question One: How can you manage GitHub Advanced Security alerts?
A) By using the GitHub Advanced Security API endpoints


Question Two: What can you do to ensure that everyone in your organization is using GitHub Advanced Security?
C) Set a security policy at the organization level.

Question Three: Which GitHub Advanced Security feature isn't available on public repositories?
B) Security Overview

Question Four: Which tool helps identify and block secrets from being pushed into repositories?
B) Secret scanning

Question Five: How can security policies be managed within GitHub?
D) All of the above

Question Six: Which of the following is NOT a step in setting security policies within GitHub?
D) Allowing unrestricted access to all users

Question Seven: How can you ensure compliance with industry standards when managing sensitive data in GitHub?
B) By using GitHub's basic security tools like secret scanning and code scanning

**Comment by ghost** [comment]
W

**Comment by ghost** [comment]
✌️

**Comment by soumya03Surya** [comment]
Posting the Answers for the questions here, covering @queenofcorgis  

<details>
<summary>Answers</summary>

### Question One:  How can you manage GitHub Advanced Security alerts?
A) By using the GitHub Advanced Security API endpoints
### Question Two: What can you do to ensure that everyone in your organization is using GitHub Advanced Security?
C) Set a security policy at the organization level.
You can enforce a security policy at the organization level that allows all repository administrators in your organization to enable features for Advanced Security for their repositories.
### Question Three: Which GitHub Advanced Security feature isn't available on public repositories?
B) Security Overview. The Security Overview is only available on private repositories with GitHub Advanced Security.
### Question Four: Which tool helps identify and block secrets from being pushed into repositories?
B) Secret scanning
### Question Five: How can security policies be managed within GitHub?
D) All of the above
### Question Six: Which of the following is NOT a step in setting security policies within GitHub?
D) Allowing unrestricted access to all users
### Question Seven: How can you ensure compliance with industry standards when managing sensitive data in GitHub?
B) By using GitHub's basic security tools like secret scanning and code scanning

</details>

**Comment by Salgado2004** [comment]
**Hello guys** 🌟👋

For this last week I just want to say I'm gonna miss you all 🥹 It was very fun to come each week studing the prep materials and creating study breakdowns to share with you (it really motivates me!).

I'm grateful for the materials, the questions, the contributions, and all. And thank you @queenofcorgis, you're an excellent moderator 😊.

Here's my last breakdown for GitHub Advanced Security:

## Domain 7: Configure GitHub Advanced Security tools in GitHub Enterprise
### GitHub Advanced Security Administration
<details>
<summary><b>Explain how GitHub Advanced Security features are enabled on GitHub Enterprise Server:</b></summary>
<p align="justify">Before enabling GHAS for GitHub Enterprise Server, first you need to enable the Advanced Security for the GH Server instance (<i>and your license <b>must support</b> Advanced security</i>). You can do that either in the interface or the CLI, but it's important to notice you have to enable each feature of Advanced Security (<i>and you'll be billed for them, so take care</i>) . After you do that, you can reload your server instance, navigate to the <b>Settings tab -> Code security and analysis</b> of your organizations and configure how you want to enable the GHAS for the organizations and repositories.</p>
</details>
<details>
<summary><b>Explain how GitHub Advanced Security features are enabled for an organization:</b></summary>
<p align="justify">To enable GHAS features in the organization level, you need to go to your organization's <b>Settings tab -> Code security and analysis</b>. You can choose to enable all the features, or only the features that you want, like dependabot. You can also check to automatically enable the features for new repositories.</p>
</details>
<details>
<summary><b>Set security policies for a repository:</b></summary>
<p align="justify">The security policies for a repository are informations displayed in the security tab of the repo (<i>for those who don't have access to alerts</i>) that tells how to report vulnerabilities, what are the supported versions of the application, and other useful stuff. To configure them, you just need to create the <b><i>security.md</b></i> markdown file in the <b><i>.github/</b></i> folder, <b><i>docs/</b></i> folder or in the root of the repository. </p>
</details>
<details>
<summary><b>Set security policies for an organization:</b></summary>
<p align="justify">The security policy in the organization level is a little more powerful. In the enterprise settings, you can create policies to allow all your organizations or only selected ones to enable GHAS features in their repositories (<i>that's smart, because it can become very expensive if uncontrolled</i>).</p>
</details>
<details>
<summary><b>Describe how permissions are interpreted throughout security workflow:</b></summary>
<p align="justify">The permissions in a security workflow are, by default, evaluated using a <b><i>GITHUB_TOKEN</b></i>. It will grant <b>read and/or write</b> permissions depending on the scope (actions, issues, PRs, etc). You can also define your own custom permissions overriding the <b><i>GITHUB_TOKEN</b></i> using the <b>permissions</b> key in workflow yaml files. It can be applied to the entire workflow or only for specific jobs. </p>
</details>
<details>
<summary><b>Locate API endpoints for GHAS features, like secret scanning, code scanning, and dependabot:</b></summary>
<p align="justify">GitHub also has different API endpoints for each Advanced Security feature, that can be useful in the development lifecycle. Each API has different purposes, but in general, they allow users to enable/disable the security feature and retrieve security alerts. The code scanning API also allows you to update SARIF files from external scanning tools. For more information, you can check the documentation in:</p>
<ol>
	<li><a href="https://docs.github.com/en/rest/reference/code-scanning">Code scanning</a></li>
	<li><a href="https://docs.github.com/en/rest/reference/secret-scanning">Secret scanning</a></li>
	<li><a href="https://docs.github.com/en/rest/reference/repos">Repos</a></li>
	<li><a href="https://docs.github.com/en/graphql">GraphQL</a></li>
</ol>
</details>
<details>
<summary><b>List stakeholders that need to be involved in the security workflows enabled by GHAS, including their role in the workflow:</b></summary>
<p align="justify">With GHAS, I understand the security workflow affects the entire development lifecycle, so this list should include from developers to product owners. The <b>developers</b> are responsible for fixing security vulnerabilities, <b>tech leaders</b> are responsible for managing the technology stack and helping the developers in the solutions, the <b>security managers</b> who create the security policies, the <b>devops engineers</b> who create the pipelines and orchestrate the security workflows and the <b>product owners</b> who manage the timelines and decide what is the standard level of security the product needs.</p>
</details>
<details>
<summary><b>Configure code scanning within a repository or organization using the default CodeQL workflow:</b></summary>
<p align="justify">Go to <b>Settings tab > Code security and analysis</b>, scroll down to code scanning section. In the <b>set up</b> drop-down you select "default". You'll be prompted some basic configurations and GitHub will do the rest for you (<i>creating a hidden workflow file that you don't have access to</i>). After you finish the setup, you can enable code scanning and it will run the first scan for the repository.</p>
</details>
<details>
<summary><b>Identify the custom build steps necessary in a CodeQL workflow:</b></summary>
<p align="justify">The initial steps are similar to the default workflow, but in the <b>code scanning section</b> you'll select "advanced". With this setup you will have a <b><i>codeql.yml</b></i> workflow file to work on. There you can configure custom build commands for your application, custom codeQL queries, file exclusions and other useful configuration. You can do it directly in the workflow file or you can reference a configuration file, like <b><i>codeql-config.yml</b></i>.</p>
</details>

### Prep questions
<details>
<summary>Answers:</summary>
<ol>
<li>A</li>
<li>C</li>
<li>B</li>
<li>B</li>
<li>D</li>
<li>D</li>
<li>B</li>
</ol>
</details>

---
Hey @profile-palash, you really inspired people with the `details` tag 😅 It's everyone now! That's awesome.

**Reply by profile-palash** [reply]
Yup, learning collaboratively was quite fun. I'll also miss this.

**Reply by queenofcorgis** [reply]
@Salgado2004 thank you for bringing so much enthusiasm to this engagement - I hope to see you and @profile-palash around the community - the fun doesn't stop here 😃

**Reply by profile-palash** [reply]
@queenofcorgis Haha absolutely, the fun and learning never stop 😊 I’m looking forward to being active in the community...

**Reply by Salgado2004** [reply]
@queenofcorgis You bet! 🚀🌟 I am also very interested in becoming active in this community 😁

**Reply by Sevc645** [reply]
aKIraH

**Comment by Pagalkhan68** [comment]
- [ ] ```

```

**Comment by bos-hieu** [comment]
I missed this week. Hopefully it is not too late.
 
- Question One: A 
- Question Two: C --> [reference](https://learn.microsoft.com/en-us/training/modules/github-administration-github-advanced-security/4-manage-access-github-advanced-security) 
- Question Three: B —> [reference](https://learn.microsoft.com/en-us/training/modules/github-administration-github-advanced-security/2-what-is-github-advanced-security) 
- Question Four: B 
- Question Five: D —> [reference](https://learn.microsoft.com/en-us/training/modules/manage-sensitive-data-security-policies/1-introduction) 
- Question Six: D 
- Question Seven: B

**Reply by queenofcorgis** [reply]
@bos-hieu not too late at all - thanks for participating!

**Comment by juanlujand** [comment]
Hi everyone

I would like to know if the certification voucher was already delivered?
Congratulations who wins this amazing benefits just in case

Have a nice week

**Reply by queenofcorgis** [reply]
@juanlujand - we just posted our [wrap up discussion ](https://github.com/orgs/community/discussions/140420) if you didn't get the ping 😃

**Comment by queenofcorgis** [comment]
Congratulations - we've completed the GHAS Certification Exam prep course. Share feedback and see if you were a voucher winner in [this Discussion](https://github.com/orgs/community/discussions/140420).

**Comment by marvaproject** [comment]
Congratulations

**Comment by vihtar** [comment]
> _This discussion and some of its comments have been edited and modified from its original format to enable evergreen learning_
> 
> This is our last part of [GHAS certification exam prep materials](https://github.com/orgs/community/discussions/137197), if you missed [Part One’s Discussion](https://github.com/orgs/community/discussions/137620), [Part Two’s Discussion](https://github.com/orgs/community/discussions/138220), and/or, [Part Three’s Discussion](https://github.com/orgs/community/discussions/138831) don’t worry feel free to get started and join us when you are caught up. The final installment covers Admin Essentials.
> 
> Plus, prep materials and test questions to continue on studying.
> 
> ## Step One: Prep 📚
> Use these materials to study before answering this week’s prep questions.
> 
> * Microsoft Learn Modules
>   
>   * [GitHub administration for GitHub Advanced Security](https://learn.microsoft.com/en-us/training/modules/github-administration-github-advanced-security/)
>   * [Manage sensitive data and security policies within GitHub](https://learn.microsoft.com/en-us/training/modules/manage-sensitive-data-security-policies/)
> * Study guides from @LadyKerr
>   
>   * [Describe GitHub Advanced Security best practices](https://github.com/LadyKerr/github-certification-guide/blob/main/study-guides/gh-advanced-security.md#domain-6-describe-github-advanced-security-best-practices-results-and-how-to-take-corrective-measures)
> * Videos
>   
>   * GitHub Advanced Security - Pass the Exam from freeCodeCamp
>     
>     * [GHAS Best Practices](https://www.youtube.com/watch?v=i740xlsqxEM&t=8260s)
>     * [GHAS for Enterprise](https://www.youtube.com/watch?v=i740xlsqxEM&t=8974s)
> * Community member materials
>   
>   * @Salgado2004's breakdown in [their comment](https://github.com/orgs/community/discussions/139526#discussioncomment-10778961)
> 
> ## Step Two: Test Your Knowledge ⚡
> ### Question One: How can you manage GitHub Advanced Security alerts?
> A) By using the GitHub Advanced Security API endpoints B) By manually checking each repository C) By using a third-party tool D) By disabling security features
> 
> ### Question Two: What can you do to ensure that everyone in your organization is using GitHub Advanced Security?
> A) Give access to security alerts to users and teams with write permission on their repositories. B) Add a SECURITY.md file to the root, docs, or .github folder of your organization's repositories. C) Set a security policy at the organization level.
> 
> ### Question Three: Which GitHub Advanced Security feature isn't available on public repositories?
> A) Secret scanning. B) Security Overview. C) Code scanning.
> 
> ### Question Four: Which tool helps identify and block secrets from being pushed into repositories?
> A) Code scanning B) Secret scanning C) Dependency management D) Issue tracking
> 
> ### Question Five: How can security policies be managed within GitHub?
> A) By creating documentation that details security guidelines B) By setting permissions and rules C) By automating processes to prevent security breaches D) All of the above
> 
> ### Question Six: Which of the following is NOT a step in setting security policies within GitHub?
> A) Creating documentation that details security guidelines B) Setting permissions and rules C) Automating processes to prevent security breaches D) Allowing unrestricted access to all users
> 
> ### Question Seven: How can you ensure compliance with industry standards when managing sensitive data in GitHub?
> A) By ignoring security alerts B) By using GitHub's basic security tools like secret scanning and code scanning C) By disabling security features D) By allowing bulk export of sensitive data
> 
> _[View the answers in this comment](https://github.com/orgs/community/discussions/139526#discussioncomment-10778893)_ 🧠
> 
> **_Use the discussion below to share additional study resources and respond to our prep questions_**

**Comment by vihtar** [comment]
апа

**Comment by mapelvader** [comment]
Q1. A
Q2. C
Q3. B
Q4. B
Q5. D
Q6. D
Q7. B

**Comment by Akash1134** [comment]
👋 Hello everyone!

We know how much you enjoyed the GHAS certification prep, so we’re thrilled to take things up a notch with the [GitHub Copilot 🤖 certification prep](https://github.com/orgs/community/discussions/144443)! 🎉

Kicking off on November 18, this four-week journey is packed with key exam topics, interactive discussions, and hands-on exercises to set you up for success. 💚

⭐ And here’s the cherry 🍒  on top: top participants will have a chance to win certification vouchers. [ 👀 Subscribe to the discussion](https://github.com/orgs/community/discussions/144443) to stay on top on what's next. 🚀
